sklearn/neighbors/_base.py

```diff
@@ -391,13 +396,15 @@ def _check_algorithm_metric(self):
         else:
             effective_p = self.p
 
-        if self.metric in ["wminkowski", "minkowski"] and effective_p < 1:
+        if self._metric in ["wminkowski", "minkowski"] and effective_p < 1:
             raise ValueError("p must be greater or equal to one for minkowski metric")
 
     def _fit(self, X, y=None):
         if self._get_tags()["requires_y"]:
             if not isinstance(X, (KDTree, BallTree, NeighborsBase)):
-                X, y = self._validate_data(X, y, accept_sparse="csr", multi_output=True)
+                X, y = self._validate_data(
+                    X, y, accept_sparse="csr", multi_output=True, order="C"
```

Comment:
This is required by the new Cython implementation. It triggers a potential memory copy in the rare cases where X is not already contiguous (even in cases when we cannot use the new Cython code (e.g. for float32 data that are not supported yet).

But as the plan will be to also write the Cython code for the 32 bit float in the future, let's keep the input data validation logic simpler by always enforcing contiguity from the start.